AsyncDisplayKit/Private/ASLayoutTransition.h

```diff
@@ -18,16 +18,45 @@
 
 @interface ASLayoutTransition : NSObject <_ASTransitionContextLayoutDelegate>
 
+/**
+ * Node to apply layout transition on
+ */
 @property (nonatomic, readonly, weak) ASDisplayNode *node;
-@property (nonatomic, readonly, strong) ASLayout *pendingLayout;
+
+/**
+ * Previous layout to transition from
+ */
 @property (nonatomic, readonly, strong) ASLayout *previousLayout;
 
-- (instancetype)initWithNode:(ASDisplayNode *)node
-               pendingLayout:(ASLayout *)pendingLayout
-              previousLayout:(ASLayout *)previousLayout;
+/**
+ * Pending layout to transition to
+ */
+@property (nonatomic, readonly, strong) ASLayout *pendingLayout;
 
+/**
+ * Returns if the layout transition can happen asynchronously
+ */
+@property (nonatomic, readonly, assign) BOOL canTransitionAsynchronous;
+
+/**
+ * Returns a newly initialized layout transition
+ */
+- (instancetype)initWithNode:(ASDisplayNode *)node pendingLayout:(ASLayout *)pendingLayout previousLayout:(ASLayout *)previousLayout NS_DESIGNATED_INITIALIZER;
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Insert and remove subnodes that where added or removed between the previousLayout and the pendingLayout
+ */
+- (void)applySubnodeTransition;
```

Comment:
I named it `startTransition` now. We should also look into finding a way to not have to expose `applySubnodeInsertions` and `applySubnodeRemovals` to the consumer.